index.js

```diff
@@ -18,54 +18,57 @@ function toCamelCase (obj) {
   return res
 }
 
-async function query (receiver) {
+class PullError extends Error {
```

Comment:
I really like this pattern of throwing an error on a failed payment that shows how much of it was completed. I think we should apply it to push payments too, though, so if you fail to send the whole amount you'll get an error containing the progress you made. This error could be renamed to something like `SPSPPaymentError`

Reply:
@wilsonianb has this already implemented. Maybe he should create the PR to do the changes to `pay`

Reply:
Would `SPSPPaymentError` be the single error used for both push and pull?
Would it just have `amount` (as opposed to `totalReceived`/`totalSent`?)

Reply:
yeah, we could do `amount`. I suppose we could also have 2 different Error classes which would be fine as well

Reply:
I like the separate errors with the explicit `totalReceived`/`totalSent`

Reply:
`pay` changes
https://github.com/interledgerjs/ilp-protocol-spsp/pull/6